clippy_lints/src/extern_without_abi.rs

```diff
@@ -0,0 +1,61 @@
+use clippy_utils::diagnostics::span_lint;
+use rustc_ast::visit::FnKind;
+use rustc_ast::{Extern, FnHeader, FnSig, ForeignMod, Item, ItemKind, NodeId};
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::declare_lint_pass;
+use rustc_span::Span;
+
+const LINT_MSG: &str = "`extern` missing explicit ABI";
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for usage of `extern` without an explicit ABI.
+    ///
+    /// ### Why is this bad?
+    /// Explicitly declaring the ABI improves readability.
```

Comment:
I think this is more of a convention than anything else, plus being in line with other tools (rustfmt)